### PR TITLE
Add GameTime  from TC 3.3.5

### PR DESCRIPTION
**Pull request by brecky**
From TC 3.3.5 add GameTime

### CODE REVIEW

**Comment by joschiwald** [comment]
i'm not going to merge this, it will be cherry picked from 3.3.5 when the time has come
your commit is incomplete aswell

File: src/server/scripts/Commands/cs_npc.cpp [hunk 1]
```diff
@@ -726,8 +727,8 @@ class npc_commandscript : public CommandScript
         uint32 displayid = target->GetDisplayId();
         uint32 nativeid = target->GetNativeDisplayId();
         uint32 Entry = target->GetEntry();
-
-        int64 curRespawnDelay = target->GetRespawnTimeEx()-time(NULL);
+
```
Comment (ForesterDev) [inline src/server/scripts/Commands/cs_npc.cpp]: Tabs

File: src/server/game/World/World.cpp [hunk 2]
```diff
@@ -2889,10 +2888,10 @@ bool World::RemoveBanCharacter(std::string const& name)
...
 {
     ///- update the time
-    time_t thisTime = time(NULL);
-    uint32 elapsed = uint32(thisTime - m_gameTime);
-    m_gameTime = thisTime;
-
+    time_t lastGameTime = GameTime::GetGameTime();
+    GameTime::UpdateGameTimers();
+    uint32 elapsed = uint32(GameTime::GetGameTime() - lastGameTime);
+
```
Comment (ForesterDev) [inline src/server/game/World/World.cpp]: Tabs

Comment (tkrokli) [reply src/server/game/World/World.cpp]: More like redundant indentation (superfluous, unneeded).

File: src/server/scripts/Commands/cs_npc.cpp (hunk 1 above)
Comment (tkrokli) [reply src/server/scripts/Commands/cs_npc.cpp]: Unneeded indentation (8 spaces).

**Comment by Treeston** [comment]
> it will be cherry picked from 3.3.5 when the time has come

**Pull request closed without merging**